### PR TITLE
Feature/copy associations

### PR DESCRIPTION
**Pull request by dav009**
We had a case in which we wanted to have all the candidate topics of a surface form associated with a different surface form.

Previously when we had similar cases we would just create a file with the associations we wanted to create. The problem with this approach is that the original `candidate counts` are lost. this  loss affects the final spot if the number of candidate topics for a single `SF` is long.

So this PR introduces:
- A way to copy candidates from sourceSurfaceForm  to destinationSurfaceForm, this copies the candidates and their counts. (added the command `copy-candidates`)
- (A fix I had in the new branch) when a new candidate topic is added to a surfaceForm in the standard way, the `candidate count` value will be the average of the `candidate counts` for the selected surfaceForm
- A refactor of the `idioCandidateStore`'s constructor, so that unit-tests could be added.
- [x] @tgalery 
- [x] rolands muffin

### CODE REVIEW

File: src/main/scala/org/idio/dbpedia/spotlight/IdioCandidateMapStore.scala [hunk 1]
```diff
@@ -7,11 +7,19 @@ package org.idio.dbpedia.spotlight
...
 import Array.concat
+import org.dbpedia.spotlight.model.{Candidate, SurfaceForm}
 
-class IdioCandidateMapStore(val pathtoFolder:String, val resStore:MemoryResourceStore){
+class IdioCandidateMapStore(var candidateMap:MemoryCandidateMapStore,val pathtoFolder:String, val resStore:MemoryResourceStore){
 
 
-  var candidateMap:MemoryCandidateMapStore = MemoryStore.loadCandidateMapStore(new FileInputStream(new File(pathtoFolder,"candmap.mem")), resStore)
+
+  def this(pathtoFolder:String, resStore:MemoryResourceStore){
```
Comment (tgalery) [inline src/main/scala/org/idio/dbpedia/spotlight/IdioCandidateMapStore.scala:16]: So this is the point were my almost null knowledge of scala is trashed. I kind of understand what you are doing here, but don't really. Maybe would be good to add a few doc strings on why this is the case?

**Comment by tgalery** [comment]
The only comments I could give would be related to refactoring the code a bit using styling conventions, which brings me to the that fact that maybe we should write a style-guide for Scala? 
However, I think this would be more of a pain when you rebase the new model editor after you merge this pr. So you are good to go.

**Comment by dav009** [comment]
I've been taking a look at: http://docs.scala-lang.org/style/

part of that job is being done in the new branch, as I refactored for quantized I've been changing the style

File: src/main/scala/org/idio/dbpedia/spotlight/IdioCandidateMapStore.scala (hunk 1 above)
Comment (SoAG) [inline src/main/scala/org/idio/dbpedia/spotlight/IdioCandidateMapStore.scala:16]: Yes agree with thiago scala is too crazy. a method this that calls this within a class???

Comment (dav009) [inline src/main/scala/org/idio/dbpedia/spotlight/IdioCandidateMapStore.scala:16]: In Scala you have a single main Constructor which is defined in the header of the class
i.e:
 `class A(val a:Int,val b:String)` states that the class `A` has a constructor receiving an int and a string.
you can add more constructors to that class but they have to reuse this base constructor, this is when `this` is used.
here `this` is  the keyword for declaring and using constructors.

So lets say you want to add  a constructor which only receives an int.
then you declare

``` scala
 class A(val a:Int,val b:String){

     # this constructor is used when only an int is given i.e: val object=new A(100) 
      def this(b:Int){
        # this is calling the base constructor
        this(b, "NoStr")  
      }
}
```

---

So basically I have two constructors:
- (i) the base constructor: which accepts a candidateStore as a parameter
- (ii) accepts the path to the CandidateStore, builds the candidateStore and pass it as a reference to the base constructor

Comment (SoAG) [inline src/main/scala/org/idio/dbpedia/spotlight/IdioCandidateMapStore.scala:16]: Ohh yeah now it makes sense :)

**Comment by SoAG** [comment]
Looks good
